v0.1.0

betzrhodes opened the pull request:
Initial beta library release

Commit:
Wrote README for existing functions

Commit:
Added new methods to be implemented to the README

Commit:
Small updates to README

Commit:
Added .gitignore

Commit:
Started new version of class

Implemented most of basic functionality
Does not support sub-second accurate timers yet

Commit:
Added device.nut and agent.nut for testing

Commit:
Removed old version of class

Commit:
Changed timer setting functions to allow for subsecond times

Haven't changed code that calls timers to allow for subsecond times yet.
Also, allowed user to pass parameters to callbacks.

Commit:
Added Makefile

Commit:
Fixed callback bug

Commit:
Made subsecond timers work

Commit:
Fixed cancelling issue

Commit:
Added new class for Jobs

Commit:
Implement pause, unpause and reset

Commit:
Changed first element of vargv to null

Commit:
Updated README

Commit:
Added empty test file

Commit:
Added Job.reset() to the README

Commit:
Updated .gitignore

Commit:
Added tests for Scheduler methods

Commit:
Added tests for Job methods

Commit:
Created separate device and agent versions of the library

Commit:
Removed unused test file

Commit:
Added Job method example to README

Commit:
Started merging device and agent into one library

Commit:
Changed device and agent files while testing

Commit:
Updated README

Commit:
Added License

Commit:
Updated .gitignore

Commit:
Updated .gitignore and removed unnecessary files

Commit:
Removed date string support

Commit:
Removed timer.class.nut

Commit:
Changed name of nowMillis to hwSec

Commit:
Removed timezone method

Commit:
Changed license to Electric Imp

Commit:
Updated README

Commit:
Fixed formatting issuesd

Commit:
Removed .imptest from .gitignore and added to git.

Commit:
Fixed README headings to match standards

Commit:
Fixed bug that appeared if a callback didn't have a bindenv.

Commit:
Combined common test code between device and agent

Commit:
Merge pull request #1 from mysticpants/master

Scheduler

Commit:
removed pp file, readme cleanup, Job now subclass

Commit:
updated test file

betzrhodes commented:
Tests are passing locally

betzrhodes commented:
Could improve test code structure, but not worth holding up library release for.

Commit:
updated test device group to use test farm

Commit:
added .travis.yml file

betzrhodes commented:
Automated testing set up - impFarm device A.

ElectricImpSampleCode committed:
Fixes

- Correct copyright date to 2018
- For at(), ensure the argument 't' is not a float (as recommended in the original read me)
- For repeatFrom(), ensure the argument 't' is not a float (as recommended in the original read me)
- Fix comments to show the correct type for certain parameters

ElectricImpSampleCode committed:
Initial edit

ElectricImpSampleCode committed:
Update README.md

ElectricImpSampleCode committed:
Update README.md

ppetrosh commented on Scheduler.lib.nut:
```diff
@@ -0,0 +1,423 @@
+    function at(t, cb, ...) {
+        local now = date();
+        local hwSec = null;
+        if (_env != ENVIRONMENT_AGENT) hwSec = hardware.millis() / 1000.0;
```
I would move this under the if/else statement below:

```
local hwSec = hardware.millis() / 1000.0;
```

This helps to make it unconditional, smaller and faster.

ppetrosh commented on Scheduler.lib.nut:
```diff
@@ -0,0 +1,423 @@
+            "args": vargv
+        });
+
+        if (_env == ENVIRONMENT_AGENT) {
```
Why is it different for device and agent?

betzrhodes replied:
Agent uses date usec for millisecond timing and and device uses hardware.millis.

Commit:
updated version number, moved hwSec var to limit scope

Commit:
added beta release note

ElectricImpSampleCode committed:
Fix some formatting

betzrhodes commented:
Working on a refactor (adding comments), so it is easier to maintain. Also cleaning up tests.

Commit:
updated at and reapeat for clarification, refactored Scheduler test files to use single helper file

Commit:
fixed merge conflicts

Commit:
update job test files

Commit:
complete code refactor, combined repeat and repeatFrom

Commit:
updated job functions

betzrhodes commented:
Code refactored and comments added.
Simplified library. Removed Job methods: reset, pause, and unpause. 
Added Job status property and getStatus method. 
Updated Job now method.
